packages/rpc-types/src/lamports.ts

```diff
@@ -46,18 +48,34 @@ export function lamports(putativeLamports: bigint): LamportsUnsafeBeyond2Pow53Mi
     return putativeLamports;
 }
 
-export function getLamportsEncoder(): FixedSizeEncoder<LamportsUnsafeBeyond2Pow53Minus1, 8> {
+export function getLamportsEncoder(
+    innerEncoder?: NumberEncoder,
+): FixedSizeEncoder<LamportsUnsafeBeyond2Pow53Minus1, 8> {
+    if (innerEncoder) {
+        return fixEncoderSize(innerEncoder, 8);
+    }
     return getMemoizedU64Encoder();
 }
 
-export function getLamportsDecoder(): FixedSizeDecoder<LamportsUnsafeBeyond2Pow53Minus1, 8> {
+export function getLamportsDecoder(
+    innerDecoder?: NumberDecoder,
+): FixedSizeDecoder<LamportsUnsafeBeyond2Pow53Minus1, 8> {
```

Comment:
I'm not sure why we are fixing the size of this codec? If I want to describe lamports using a u32, this will still use 8 bytes which I think is wrong. Unless I'm missing something?

Reply:
Okay cool, I'll loosen the output type here to `NumberDecoder` and we can remove the fix. I thought the `FixedSizeDecoder<LamportsUnsafeBeyond2Pow53Minus1, 8>` was important in this signature and didn't want to break it!

Reply:
Ideally this should keep the size of the provided number codec but map the number type to a Lamport type.

Reply:
Reworked this to use new functions like this:

```
export function getLamportsEncoderFromEncoder<TEncoder extends NumberEncoder>(
    innerEncoder: TEncoder,
): Encoder<LamportsUnsafeBeyond2Pow53Minus1> & ExtractAdditionalProps<TEncoder, NumberEncoder> {
    return innerEncoder;
}
```

Like you said, the version without a wrapped encoder is used all over the place, so best to keep it small and tree-shake this stuff away

And then when there is a wrapped encoder, it no longer gets its size changed, and the `size` remains on the lamport encoder type too.